lib/src/rules/file_names.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/utils.dart';
+
+const _desc = r'Name source files using `lowercase_with_underscores`.';
+
+const _details = r'''
+
+**DO** name source files using `lowercase_with_underscores`.
+
+Some file systems are not case-sensitive, so many projects require filenames
+to be all lowercase. Using a separate character allows names to still be
```

Comment:
separate => separator?

Reply:
I updated the sentence with the content of https://www.dartlang.org/guides/language/effective-dart/style#do-name-libraries-and-source-files-using-lowercase_with_underscores